betty/gramps/loader.py

```diff
@@ -1175,3 +1242,432 @@ def _load_attributes_for(
+                    person_count=await self._load_people()
+                )
+            )
+
+            await self._load_families()
+
+        resolve(*self._ancestry)
+
+    async def _load_notes(self) -> int:
+        handles = self._database.get_note_handles()
+        await gather(*map(self._load_note, handles))
+        return len(handles)
+
+    async def _load_note(self, handle: str) -> None:
+        gramps_note = cast("GrampsNote", self._database.get_note_from_handle(handle))
+        note = Note("Hello, world!", id=gramps_note.gramps_id)
+        self._load_entity(note, gramps_note)
+
+    async def _load_medias(self) -> int:
+        handles = self._database.get_media_handles()
+        await gather(*map(self._load_media, handles))
+        return len(handles)
+
+    async def _load_media(self, handle: str) -> None:
+        gramps_media = cast("GrampsMedia", self._database.get_media_from_handle(handle))
+        file = File(Path(__file__), id=gramps_media.gramps_id)
+        self._load_entity(file, gramps_media)
+
+    async def _load_repositories(self) -> int:
+        handles = self._database.get_repository_handles()
+        await gather(*map(self._load_repository, handles))
+        return len(handles)
+
+    async def _load_repository(self, handle: str) -> None:
+        gramps_repository = cast(
+            "GrampsRepository", self._database.get_repository_from_handle(handle)
+        )
+        source = Source(id=gramps_repository.gramps_id)
+        self._load_entity(source, gramps_repository)
+
+    async def _load_sources(self) -> int:
+        handles = self._database.get_source_handles()
+        await gather(*map(self._load_source, handles))
+        return len(handles)
+
+    async def _load_source(self, handle: str) -> None:
+        gramps_source = cast(
+            "GrampsSource", self._database.get_source_from_handle(handle)
+        )
+        source = Source(id=gramps_source.gramps_id)
+        self._load_entity(source, gramps_source)
+
+    async def _load_citations(self) -> int:
+        handles = self._database.get_citation_handles()
+        await gather(*map(self._load_citation, handles))
+        return len(handles)
+
+    async def _load_citation(self, handle: str) -> None:
+        gramps_citation = cast(
+            "GrampsCitation", self._database.get_citation_from_handle(handle)
+        )
+        citation = Citation(id=gramps_citation.gramps_id, source=Source())
+        self._load_entity(citation, gramps_citation)
+
+    async def _load_places(self) -> int:
+        handles = self._database.get_place_handles()
+        await gather(*map(self._load_place, handles))
+        return len(handles)
+
+    async def _load_place(self, handle: str) -> None:
+        gramps_place = cast("GrampsPlace", self._database.get_place_from_handle(handle))
+        place = Place(id=gramps_place.gramps_id)
+        self._add_entity(place, gramps_place)
+
+    async def _load_events(self) -> int:
```

Comment:
Combine all of these into a _load_handles() method

Reply:
That would not actually decrease the amount of code because ngettext() still needs to be called with the actual handle count.